vllm/engine/multiprocessing/client.py

```diff
@@ -685,23 +686,37 @@ async def reset_prefix_cache(self) -> None:
             request=RPCResetPrefixCacheRequest.RESET_PREFIX_CACHE,
             socket=self.input_socket)
 
-    async def add_lora(self, lora_request: LoRARequest) -> None:
-        """Load a new LoRA adapter into the engine for future requests."""
-        # Uses the same I/O as generate requests
-        request = RPCLoadAdapterRequest(lora_request)
-
+    async def send_request_and_get_response(self, request: Any) -> Any:
         # Create output queue for this requests.
         queue: asyncio.Queue[Union[None, BaseException]] = asyncio.Queue()
-        self.output_queues[request.request_id] = queue
+        self.output_queues[id(request)] = queue
```

Comment:
@youkaichao this won't work - the request_id is used for correlation of responses that are later sent back from the engine... `id()` just gives the address of the request object in memory.

Generalization of this req/response will need changes in a couple more places like here https://github.com/vllm-project/vllm/blob/main/vllm/engine/multiprocessing/client.py#L249